tests/utils/secrets.go

```diff
@@ -48,3 +48,14 @@ func CreateSecretFromDefinition(c *kubernetes.Clientset, secret *v1.Secret) (*v1
 	}
 	return secret, nil
 }
+
+func DeleteSecret(clientSet *kubernetes.Clientset, namespace string, secret v1.Secret) error {
+	e := wait.PollImmediate(SecretPollInterval, SecretPollPeriod, func() (bool, error) {
+		err := clientSet.CoreV1().Secrets(namespace).Delete(secret.GetName(), nil)
```

Comment:
don't worry about `err` being assigned here. This copy of `err` is referred to as a _shadow_ variable and is its own instance of error.